### PR TITLE
feat: adds BigQuery parameter samples

### PR DESCRIPTION
**Pull request by telpirion**
This adds the following region tags:

- bigquery_query_params_timestamps
- bigquery_query_params_arrays
- bigquery_query_params_named
- bigquery_query_params_positional

These samples to be published here:
https://cloud.google.com/bigquery/docs/parameterized-queries

### CODE REVIEW

**Comment by amanda-tarafa** [comment]
@telpirion  I'll get to these by mid week unless they are urgent. I want to fix possible issues caused by #952 first.

**Comment by telpirion** [comment]
These aren't urgent--just cleaning up some work items in our backlog. Thank you for the update!

File: bigquery/api/Snippets/QueryWithTimestampParameters.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+public class BigQueryQueryWithTimestampParameters
+{
+    public void QueryWithTimestampParameters(
+        string projectId = "your-project-id"
+    )
+    {
+        var timestamp = new DateTime(2016, 12, 7, 8, 0, 0, DateTimeKind.Utc);
+
+        // Note: Standard SQL is required to use query parameters.
+        var query = "SELECT TIMESTAMP_ADD(@ts_value, INTERVAL 1 HOUR);";
```
Comment (amanda-tarafa) [inline bigquery/api/Snippets/QueryWithTimestampParameters.cs]: Ohhh, can't you make a query against a table that has a DataTime column and maybe show something like `where date > @ts_value and date < TIMESTAMP_ADD(@ts_value, INTERVAL 1 HOUR)`?
It's just that using BigQuery to add an hour to a DateTime doesn't look like best practices we should be showcasing.

Comment (telpirion) [reply bigquery/api/Snippets/QueryWithTimestampParameters.cs]: :) 

I'm not a BigQuery expert, but it's what they do in their canonical:
https://github.com/googleapis/python-bigquery/blob/dc389498391798235383fbec20e48a23128ddc6a/samples/client_query_w_timestamp_params.py#L27

File: bigquery/api/Snippets/QueryWithArrayParameters.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+//
+// [START bigquery_query_params_arrays]
+
+using System;
+using Google.Cloud.BigQuery.V2;
+
```
[the remark this answers is not in the document]
Comment (telpirion) [reply bigquery/api/Snippets/QueryWithArrayParameters.cs]: Done.

File: bigquery/api/Snippets/QueryWithArrayParameters.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+public class BigQueryQueryWithArrayParameters
+{
+    public void QueryWithArrayParameters(
+        string projectId = "your-project-id"
+    )
```
[the remark this answers is not in the document]
Comment (telpirion) [reply bigquery/api/Snippets/QueryWithArrayParameters.cs]: Done.

File: bigquery/api/Snippets/QueryWithArrayParameters.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+// [START bigquery_query_params_arrays]
+
+using System;
+using Google.Cloud.BigQuery.V2;
+
+
+public class BigQueryQueryWithArrayParameters
+{
+    public void QueryWithArrayParameters(
+        string projectId = "your-project-id"
```
[the remark this answers is not in the document]
Comment (telpirion) [reply bigquery/api/Snippets/QueryWithArrayParameters.cs]: Done (here).

File: bigquery/api/Snippets/QueryWithArrayParameters.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+//
+// [START bigquery_query_params_arrays]
+
+using System;
```
[the remark this answers is not in the document]
Comment (telpirion) [reply bigquery/api/Snippets/QueryWithArrayParameters.cs]: Done.

**Comment by telpirion** [comment]
I tried to put all of the method params all on the same line as the rest of the method signature, but some of the results were longer than 80 chars.

File: bigquery/api/Snippets/QueryWithPositionalArguments.cs [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+// [START bigquery_query_params_positional]
+
+using Google.Cloud.BigQuery.V2;
+using System;
+
+public class BigQueryQueryWithPositionalParameters
+{
+    public void QueryWithPositionalParameters(
+        string projectId = "your-project-id"
+    )
```
Comment (amanda-tarafa) [inline bigquery/api/Snippets/QueryWithPositionalArguments.cs]: nit: Even if the parameters don't fit in one line, this parentheses should be right after the last parameter. Never by itself.

File: bigquery/api/Snippets/QueryWithTimestampParameters.cs [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+// [START bigquery_query_params_timestamps]
+
+using Google.Cloud.BigQuery.V2;
+using System;
+
+public class BigQueryQueryWithTimestampParameters
+{
+    public void QueryWithTimestampParameters(
+        string projectId = "your-project-id"
+    )
```
Comment (amanda-tarafa) [inline bigquery/api/Snippets/QueryWithTimestampParameters.cs]: nit: same here, parentheses always after the last parameter.

File: bigquery/api/Snippets/QueryWithTimestampParameters.cs (hunk 1 above)
Comment (amanda-tarafa) [reply bigquery/api/Snippets/QueryWithTimestampParameters.cs]: Hmm, leave as is then. I'll try and ask them about changing the sample.

File: bigquery/api/Snippets/QueryWithTimestampParameters.cs (hunk 7 above)
Comment (telpirion) [reply bigquery/api/Snippets/QueryWithTimestampParameters.cs]: Done.

File: bigquery/api/Snippets/QueryWithPositionalArguments.cs (hunk 6 above)
Comment (telpirion) [reply bigquery/api/Snippets/QueryWithPositionalArguments.cs]: Done.

**Comment by amanda-tarafa** [comment]
Mergin, CI is knowingly broken but unrelated to this PR.

**Pull request merged**